src/darkslide/generator.py

```diff
@@ -505,6 +507,44 @@ def register_macro(self, *macros):
                 raise TypeError("Couldn't register macro; a macro must inherit"
                                 " from macro.Macro")
 
+    def embed_url_data(self, context, html):
+        """Find all image and fonts referenced in CSS with an ``url()`` function
+        and embed them in base64. Images from the user (i.e. included in its
+        source code, *not* in its CSS) are embedded by the macro
+        `EmbedImagesMacro`.
+        """
+        all_urls = re.findall(r'url\([\"\']?(.*?)[\"\']?\)', html,
+                              re.DOTALL | re.UNICODE)
+        embed_exts = ('.jpg', '.jpeg', '.png', '.gif', '.svg', '.woff2',
+                      '.woff')
+        embed_urls = (url for url in all_urls if url.endswith(embed_exts))
+
+        css_dirs = (
+            [os.path.join(self.theme_dir, 'css')] +
+            [css_entry['dirname'] for css_entry in context['user_css']]
+        )
+
+        for embed_url in embed_urls:
+            embed_url = embed_url.replace('"', '').replace("'", '')
+
+            def _try_to_encode(embed_url):
```

Comment:
I think this can simply be replaced with a for/break/else pattern instead.

Reply:
You're right... Talk about over-engineering.
Thanks, will fix.